### ISSUE TITLE
Date validation returns different results whether providing a format

### ISSUE THREAD

**Issue by oscarotero** [Respect/Validation#1083]
Hi!
As you can see in the following code: 

```php
v::date()->validate('2014-99'); //false
v::date('Y-m')->validate('2014-99'); //true
```
The date validator with a format returns `true` with an invalid date. It's due [the use of `date_parse_from_format`](https://github.com/Respect/Validation/blob/1.1/library/Rules/Date.php#L52) that only parse the string according to a format but without check whether the date is valid or not. Why not use [`strtotime`](https://github.com/Respect/Validation/blob/1.1/library/Rules/Date.php#L40) function in that case too?

**Comment by henriquemoody** [comment]
Thanks for reporting, @oscarotero! 🐼 

We used to use `DateTime::createFromFormat()` but we used to get quite a few false positives as well (https://3v4l.org/A0onA). The `strtotime()` guess way too much and we also get some false-positives and vice-versa. I think we need we can also use [checkdate](https://php.net/checkdate) as an addition to the result of `date_parse_from_format()`.

You can see the reason why we changed it in eecc696792ea7ced66d6bbdbe08247e31d6ba951

**Comment by oscarotero** [comment]
I see.
To me, `strtotime` provides better results, at least in my use cases. Currently I've fixed this issue using the validator twice:

```php
v::date()->date('Y-m');
```
I need also validate time values (like `18:30`) so `checkdate` cannot help here.

IMHO, `date()` and `date($format)` should perform the same date validation, with the addition of `date_parse_from_format` in the latest one.

**Comment by henriquemoody** [comment]
`strtotime` is great, but because it tries to guess too much sometimes it can be very disappointing. For example, the date `2012-31-12` is valid (`Y-d-m`), however `strtotime` returns false for it, that's why we shouldn't use `strtotime` as a pre-condition for the rule.

That said, if you don't define the format  the rule will try to _guess_ what's the format and sometimes it will blow up the validation. If you pass the format it needs to work properly. 

I fixed the problem when you define the format, please update your Validation to the version 1.1.24.

Thanks for reporting! 🐼

**Comment by oscarotero** [comment]
Thank you for the quick fix 👍
